### PR TITLE
Update API to disable short_name update

### PR DESCRIPTION
**Pull request by mariarivera30**

### CODE REVIEW

File: pybossa/api/api_base.py [hunk 1]
```diff
@@ -420,6 +420,7 @@ def _update_instance(self, existing, repo, repos, new_upload=None):
             new_data = request.form
         self._forbidden_attributes(new_data)
         self._restricted_attributes(new_data)
+        self._forbidden_attributes_put(new_data)
```
Comment (fiorda) [inline pybossa/api/api_base.py]: There is a `immutable_attributes` property. Is that something we could have used?

File: pybossa/api/project.py [hunk 2]
```diff
@@ -89,8 +90,15 @@ def _forbidden_attributes(self, data):
...
                 if key == 'published':
-                    raise Forbidden('You cannot publish a project via the API')
+                    raise Forbidden('You cannot publish a project via the API')   
                 raise BadRequest("Reserved keys in payload")
+    
+    def _forbidden_attributes_put(self, data):
+        for key in data.keys():
+            if key in self.put_reserved_keys:
+                if key == 'short_name':
+                    raise Forbidden('You cannot update short_name via the API')
```
Comment (fiorda) [inline pybossa/api/project.py]: Do we need the special handling? Or would `raise Forbidden('Cannot change {} via API'.format(key))` work?

Comment (mariarivera30) [reply pybossa/api/project.py]: No special handling is needed its correct.  Deep suggested to not create a new function and instead use request.method in _forbidden_attributes() Im planning to do it that way.

File: pybossa/forms/forms.py [hunk 3]
```diff
@@ -99,6 +99,7 @@ class ProjectUpdateForm(ProjectForm):
                                 message=lazy_gettext(
                                     "You must provide a description.")),
                              validators.Length(max=255)])
+    short_name = TextField(label=None, widget=HiddenInput())
```
Comment (fiorda) [inline pybossa/forms/forms.py]: I think that if this is left here, `short_name` can still be changed via `/project/<short_name>/update`

Comment (mariarivera30) [reply pybossa/forms/forms.py]: If that is the case then id can be changed too? line 96

Comment (mariarivera30) [reply pybossa/forms/forms.py]: if that is the case then  id can be changed too? line 96

Comment (fiorda) [reply pybossa/forms/forms.py]: No, sorry - I didn't see https://github.com/bloomberg/pybossa/pull/190/files#diff-8fa3bf883c1a925427ca6a3b4587f452L603

Comment (mariarivera30) [reply pybossa/forms/forms.py]: cool!

**Pull request closed without merging**